src/main/java/hudson/plugins/translation/L10nDecorator.java

```diff
@@ -97,9 +100,16 @@ public String encodeRecording(StaplerRequest request) throws IOException {
      * Does the opposite of {@link #encodeRecording(StaplerRequest)}.
      */
     public List<Msg> decode(StaplerRequest request) throws IOException {
-        BufferedReader r = new BufferedReader(new InputStreamReader(new GZIPInputStream(new CipherInputStream(
+        final GZIPInputStream gzipInpurStream = new GZIPInputStream(new CipherInputStream(
                 new ByteArrayInputStream(Base64.decode(request.getParameter("bundles").toCharArray())),
-                getCipher(DECRYPT_MODE)))));
+                getCipher(DECRYPT_MODE)));  
+        final BufferedReader r;
+        try {
+            //TODO: Replace by StandardCharsets in JDK7
+            r = new BufferedReader(new InputStreamReader(gzipInpurStream, "UTF-8"));
+        } catch (UnsupportedEncodingException ex) {
+            throw new IOException("UTF-8 encoding is not supported");
```

Comment:
Message should clarify that UTF-8 support is required by the JVM runtime contract and as such something is very very wrong if this exception gets thrown... in fact it is really an illegal state for this exception to be thrown